Inspector/macho.c

```diff
@@ -0,0 +1,378 @@
+    uint32_t command_count = get_load_command_count(mach_hdr);
+    const struct load_command **load_commands = get_load_commands(mach_hdr);
+    uint32_t type_count = get_load_command_count_by_type(mach_hdr, type);
+    const struct load_command **type_commands = malloc(type_count * sizeof(struct load_command *));
+    uint32_t type_index = 0;
+    for (int i = 0; i < command_count; i++) {
+        const struct load_command *cmd = load_commands[i];
+        if (cmd->cmd == type) {
+            type_commands[type_index] = cmd;
+            type_index++;
+        }
+    }
+    free(load_commands);
+    
+    return type_commands;
+}
+
+const char *get_name(uint32_t image_no) {
+    return _dyld_get_image_name(image_no);
+}
+
+const uint8_t *get_uuid(uint32_t image_no) {
+    const struct mach_header *mach_hdr = _dyld_get_image_header(image_no);
+    uint32_t command_count = get_load_command_count_by_type(mach_hdr, LC_UUID);
+    const struct load_command **load_commands = get_load_commands_by_type(mach_hdr, LC_UUID);
+    for (int i = 0; i < command_count;) {
+        return ((const struct uuid_command *)load_commands[i])->uuid;
+    }
+    free(load_commands);
+    
+    return NULL;
+}
+
+uint8_t *get_cs_blob_by_magic(uint32_t image_no, enum cs_magic magic) {
+    const struct mach_header *mach_hdr = get_mach_header(image_no);
+    uint32_t command_count = get_load_command_count_by_type(mach_hdr, LC_CODE_SIGNATURE);
+    const struct load_command **load_commands = get_load_commands_by_type(mach_hdr, LC_CODE_SIGNATURE);
+    for (int i = 0; i < command_count; i++) {
+        const struct linkedit_data_command *data_command = (const struct linkedit_data_command *)load_commands[i];
+        uint32_t signature_offset = data_command->dataoff;
+        
+        uint8_t *lc_code_signature = (uint8_t *)mach_hdr + signature_offset;
+        CS_SuperBlob *super_blob = (CS_SuperBlob *)lc_code_signature;
+        if (ntohl(super_blob->magic) == CSMAGIC_EMBEDDED_SIGNATURE) {
+            uint32_t blob_count = ntohl(super_blob->count);
+            for (int i = 0; i < blob_count; i++) {
+                uint32_t blob_offset = ntohl(super_blob->index[i].offset);
+                uint8_t *blob_bytes = lc_code_signature + blob_offset;
+                CS_Blob *blob = (CS_Blob *)blob_bytes;
+                uint32_t blob_magic = ntohl(blob->magic);
+                if (blob_magic == magic) {
+                    return blob_bytes;
+                }
+            }
+        }
+    }
+
+    return NULL;
+}
+
+uint8_t *get_entitlements_blob(uint32_t image_no) {
+    return get_cs_blob_by_magic(image_no, CSMAGIC_ENTITLEMENTS);
+}
+
+char *get_entitlements_plist(uint32_t image_no) {
+    CS_Blob *blob = (CS_Blob *)get_entitlements_blob(image_no);
+    uint32_t plist_length = ntohl(blob->length) - 8;
+    char *plist = malloc(sizeof(char) * (plist_length + 1));
+    memset(plist, 0, plist_length);
```

Comment:
Why clear to 0 if you are going to overwrite it all immediately? (And why clear to zero rather than calloc()?)